executor/cook/progress.py

```diff
@@ -20,78 +20,58 @@ def increment_and_get(self):
 
 
 class ProgressUpdater(object):
-    """This class is responsible for sending progress updates to the scheduler.
-    It throttles the rate at which progress updates are sent.
-    """
+    """This class is responsible for sending progress updates to the scheduler."""
 
-    def __init__(self, task_id, max_message_length, poll_interval_ms, send_progress_message_fn):
+    def __init__(self, task_id, max_message_length, send_progress_message_fn):
         """
         task_id: string
             The task id.
         max_message_length: int
             The allowed max message length after encoding.
-        poll_interval_ms: int
-            The interval after which to send a subsequent progress update.
         send_progress_message_fn: function(message)
             The helper function used to send the progress message.
         """
         self.task_id = task_id
         self.max_message_length = max_message_length
-        self.poll_interval_ms = poll_interval_ms
-        self.last_reported_time = None
         self.last_progress_data = None
         self.send_progress_message = send_progress_message_fn
         self.lock = Lock()
 
-    def has_enough_time_elapsed_since_last_update(self):
-        """Returns true if enough time (based on poll_interval_ms) has elapsed since 
-        the last progress update (available in last_reported_time).
-        """
-        if self.last_reported_time is None:
-            return True
-        else:
-            current_time = time.time()
-            time_diff_ms = (current_time - self.last_reported_time) * 1000
-            return time_diff_ms >= self.poll_interval_ms
-
-    def send_progress_update(self, progress_data, force_send=False):
+    def send_progress_update(self, progress_data):
         """Sends a progress update if enough time has elapsed since the last progress update.
-        The force_send flag can be used to ignore the check for enough time having elapsed.
         Using this method is thread-safe.
+        This method will trim the message if it is too large.
         
         Parameters
         ----------
         progress_data: dictionary
             The progress data to send.
-        force_send: boolean, optional
-            Defaults to false.
-            
+
         Returns
         -------
         Nothing 
         """
         with self.lock:
             if progress_data is not None and self.last_progress_data != progress_data:
-                if force_send or self.has_enough_time_elapsed_since_last_update():
-                    logging.info('Sending progress message {}'.format(progress_data))
-                    message_dict = dict(progress_data)
-                    message_dict['task-id'] = self.task_id
+                logging.info('Sending progress message {}'.format(progress_data))
+                message_dict = dict(progress_data)
+                message_dict['task-id'] = self.task_id
+                progress_message = json.dumps(message_dict)
```

Comment:
Why is the maximum length checked on the JSON object, but we truncate the message in the object? It seems like it would make more sense to check the message length on the message itself.

Reply:
We chose to interpret the max length to refer to the un-encoded version.

The message itself is encrypted and we can't guarantee lengths of the encrypted version by trimming the message.